### DISCUSSION TITLE
PWM for Airjet Speed?

### DISCUSSION THREAD

**Discussion by Addonnis** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#195] [Ideas]
I've added a PWM inline with the blower motor - This allows physical adjustment of the airjet intensity which is great for lower noise, power consumption and heat loss when in use. 

Is there a digital PWM I could add that could be controlled by the ESP8266? Would it be possible in theory to use D0, D8, 3V3 and GND to control bubble intensity, or are these 'input only'?

Thanks in advance!

![271660620_335924508395378_2995080467517923688_n](https://user-images.githubusercontent.com/97511696/150625131-6bf6211e-88aa-4e0c-b59c-99b524923d78.jpg)
![271868711_301380935373424_435274030788183411_n](https://user-images.githubusercontent.com/97511696/150625134-97ea8fee-b4a7-407e-8934-5f2c458c20c6.jpg)

**Comment by visualapproach** [comment]
It's possible in theory. You'd need to make an interface between the pins and the controlled thing. The pin output can only source a few mA. A PWM output is probably the way to go. I think the 8266 is doing PWM in software so it might affect the rest of the code speed wise. Then you'd need to do some code to handle user input and control output. And test if it also works in practice.
